subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DefaultModuleVersionIdentifier.java

```diff
@@ -21,19 +21,20 @@
 
 public class DefaultModuleVersionIdentifier implements ModuleVersionIdentifier {
 
-    private final DefaultModuleIdentifier id;
+    private final ModuleIdentifier id;
     private final String version;
 
     public DefaultModuleVersionIdentifier(String group, String name, String version) {
```

Comment:
Yes, at some point it would be good to do this. However this is not a performance problem as of now.